src/agent/src/sandbox.rs

```diff
@@ -233,14 +233,29 @@ impl Sandbox {
             online_memory(&self.logger)?;
         }
 
-        let cpuset = rustjail_cgroups::fs::get_guest_cpuset()?;
+        let guest_cpuset = rustjail_cgroups::fs::get_guest_cpuset()?;
 
         for (_, ctr) in self.containers.iter() {
+            let cpu = ctr
+                .config
+                .spec
+                .as_ref()
+                .unwrap()
+                .linux
+                .as_ref()
+                .unwrap()
+                .resources
+                .as_ref()
+                .unwrap()
+                .cpu
+                .as_ref();
+            let container_cpust = if let Some(c) = cpu { &c.cpus } else { "" };
+
             info!(self.logger, "updating {}", ctr.id.as_str());
             ctr.cgroup_manager
                 .as_ref()
                 .unwrap()
-                .update_cpuset_path(cpuset.as_str())?;
+                .update_cpuset_path(guest_cpuset.as_str(), &container_cpust)?;
```

Comment:
Just to double check my understanding:

 Today the guest doesn't handle CPUSets for containers. If a static policy is created, and a specific cpuset is set for the pod's containers, it will not be applied for the container, but instead the container itself will inherit the guest-cpuset?

Reply:
I didn't think I was using a static policy when observing #1156.

Reply:
Was the issue specifically around https://github.com/kata-containers/kata-containers/pull/1158/files#diff-4d3be83722e16562f9007284c009e030d0d72d382c1220dd4280b0dd75034c51R368 ?

Reply:
Hi @egernst, sorry for confusing, this PR includes two fixes:

- one is https://github.com/kata-containers/kata-containers/pull/1158/files#diff-4d3be83722e16562f9007284c009e030d0d72d382c1220dd4280b0dd75034c51R368-R370 , this will lead starting container failed if `cpuset` is set
- one is for #1156 , the container only use one CPU because the cpu is not set, this is fixed by https://github.com/kata-containers/kata-containers/pull/1158/files#diff-4d3be83722e16562f9007284c009e030d0d72d382c1220dd4280b0dd75034c51L1047-R1049 , this will update container's cpuset, first it will inherit the guest's cpuset.